### PR TITLE
[Z2 MAP] Centcom Response Team Variations

### PR DESCRIPTION
**Pull request by Ikarrus**
- The button is now labelled "Make Centcom Response Team"
- Added different response teams of varying strength, to give admins more options than balls-to-the-walls all the time
- Admins get to pick from the following teams to send in:
  - Code Green: Centcom Official
  - Code Blue: Light ERT - no access to lockers or admin weapons
  - Code Amber: Full ERT - access to lockers, but not admin weapons
  - Code Red: Elite ERT - what we have now.
  - Code Delta: Deathsquad'
- Squad size defined by the admin. 1-7 people
- Removed Make Deathsquad button as it's all collapsed into this one button now
- All centcom response teams spawn in the same situation room
- Also, 00 get

### CODE REVIEW

**Comment by Fayrik** [comment]
Doesn't add the ability to chose the squad size.
Doesn't add the ability to chose extra equipment.
Removes the ability to chose which kind of squad you're sending in.
Map edit without a screenshot.
:-1: I'd say what we've got already is marginally better.

**Comment by Ikarrus** [comment]
> Removes the ability to chose which kind of squad you're sending in.

We don't have that?

**Comment by Fayrik** [comment]
We have the ability to choose between a Deathsquad or an ERT.
And now the Deathsquad's prerequisite is literally impossible to obtain.

File: code/modules/admin/verbs/one_click_antag.dm [hunk 1]
```diff
@@ -343,6 +342,8 @@ client/proc/one_click_antag()
 				equip_deathsquad(Commando)
 			Commando.key = chosen_candidate.key
 			Commando.mind.assigned_role = "Death Commando"
+			for(var/obj/machinery/door/poddoor/ert/door in world)
```
Comment (RemieRichards) [inline code/modules/admin/verbs/one_click_antag.dm]: > in world

Pretty sure doors or atleast machines have a list.

File: code/modules/admin/verbs/one_click_antag.dm [hunk 2]
```diff
@@ -438,28 +496,33 @@ client/proc/one_click_antag()
...
 				if(7)
 					ERTOperative.real_name = "Engineer [pick(lastname)]"
-					equip_emergencyresponsesquad(ERTOperative, "eng")
+					equip_emergencyresponsesquad(ERTOperative, "eng",redalert)
 			ERTOperative.key = chosen_candidate.key
 			ERTOperative.mind.assigned_role = "ERT"
 
+			//Open the Armory doors
+			if(alert != "Blue")
+				for(var/obj/machinery/door/poddoor/ert/door in world)
```
Comment (RemieRichards) [inline code/modules/admin/verbs/one_click_antag.dm]: see above.

**Comment by tedward1337** [comment]
What we have now is fine. ERT is never spawned in as often as people think. No reason to change it.

**Comment by Ikarrus** [comment]
@fayrik I still have no idea what you're talking about? If you want to have more control over it I honestly believe it would be faster and easier just to do it manually than fiddling with some awful html panel/popups. 

@tedward nothing is being changed. Just added to.

**Comment by MrStonedOne** [comment]
@Fayrik @tedward1337 @Ikarrus 

This doesn't work based off the station alert system. You get to pick what alert level to spawn in when you spawn them.

**Comment by Ikarrus** [comment]
Yeah you guys are completely misunderstanding what this actually is.

**Comment by Fayrik** [comment]
Then using the same "codes" as the station is very misleading.
In which case, this is actually the improvement I had expected.
The armoury area blocks the ERT's view of the shuttle though, something I had planned due to the ability to spawn multiple ERTs, meaning they could end up in the office without a shuttle.
Also, why are you limiting ERTs to 5 people?
It goes from 4 to 7 because of the available job roles, but limiting the maximum down to 5 feels like it's just going to remove people's chances of getting in on the fun.

**Comment by Alek2ander** [comment]
Speaking of code reds, admins are long overdue for a button that changes the station alert level

File: code/modules/admin/verbs/one_click_antag.dm (hunk 1 above)
Comment (GunHog) [inline code/modules/admin/verbs/one_click_antag.dm]: Use the "airlocks" list.

**Comment by Cheridan** [comment]
Considering that a higher-pop round on Sibyl is 70 players, a team of 7 is 10% of the entire station. It's a small army in the context of this game.

**Comment by xxalpha** [comment]
Maybe numbers can be tweaked, but this is fucking great.

:+1:

**Comment by Miauw62** [comment]
Just add an extra input for squad size, it's convenient.
Op 27-apr.-2015 16:18 schreef "xxalpha" <notifications@github.com>:

**Comment by Ikarrus** [comment]
@Miauw62 Aren't we supposed to be stepping away from popups galore?

If I add another one it will be 3-4 popups after another

**Comment by Fayrik** [comment]
I was just curious why you chose 5, 7 could be a bit excessive, but 5 would leave you with 2 security officers where as 4 would only give you one of each job. 6 is odd, but at least with the extra engineer that second security member isn't feeling quite so odd on his own.
It strikes me that the best solution to this would be to swap the medic position with the security position, so that there isn't instantly an overpopulation of security for every mission, but sticking to 4, 7, 10, 13 and on means that regardless of the situation, the squad is well rounded for any situation.
But so long as we're not just stuffing the ERT full of security, it's not really an issue.

The map however, is really getting to me. Could you please just move the armoury 6 tiles to the left? It's current position is horrendously jarring.

**Comment by Ikarrus** [comment]
>horrendously jarring

Nice opinion, could you explain how?

I don't see a problem with the map, the armory is where it is so the squad walks by it on the way to the shuttle, and they get to see the shuttle while they equip themselves.

**Comment by Fayrik** [comment]
As I said before, spawning an ERT doesn't necessarily mean the shuttle is there, and considering those doors only open half the time, that means the window is only visible half the time.
Not to mention the fact that the corridor actually lengthens the walk to the shuttle anyway, meaning regardless of if it's open, everyone's gotta trudge past it.

Plus if they're equipping themselves by the time the commander's noticed the shuttle's missing, then the pray/adminhelp to get it back will be sent by the time they're already ready.

I'd specifically designed the whole area to be high visibility, however, and by putting a solid wall in on the armoury and removing that visibility, you're basically putting the middle finger up to my whole design.

**Comment by Ikarrus** [comment]
I swear there used to be a shuttle console on centcom for that very reason.

I'll see about re-adding that.

**Comment by Steelpoint** [comment]
I think the proposed system is good. It allows admins to scale Centcom's response to a particular situation.

The current system, while good, operates under the assumption that the station has gone to hell and back. Which while that is usually the scenario where the ERT are called in, it however does not scale well to any other scenario.

However I would like to know the definition of 'admin weapons'? Does this mean Pulse Carbines or what? Because the Pulse weapons are the ERT's standard issue.

**Comment by Ikarrus** [comment]
I asked adminbus at the time and they felt pulse weapons were far too powerful, and I agree. They are swapped with more standard NT energy weapons in lower distress levels.

**Comment by Steelpoint** [comment]
Ehh I don't know, I did set up the ERT's Pulse Weapons with Loyalty Firing Pins to prevent the crew from looting them, also they don't get a huge amount of ammunition for them when blasting away on Pulse mode.

Normal Energy weapons really don't suit the job well, at least for anything higher than then proposed Code Blue: Light level of ERT.

**Comment by Ikarrus** [comment]
I think it's the fact that it kills in two hits and can destroy walls makes it overpowered, no matter how much ammo you give it.

**Comment by Steelpoint** [comment]
Technically its a four hit kill. Besides, the pulse setting gives the ERT versatility to move around in a hostile situation.

While I would agree that no Pulse weapons on a light mission seems fine, I would disagree with withholding Pulse weapons on a full mission, if anything it seems wierd that the only difference between a Elite and a Full mission is that one team has Pulse weapons while the other does not, yet I would highly doubt the presence of Pulse weapons very heavily changes the capabilities of the team. 

They still have access to their other, very robust, equipment such as the far more powerful Defib units or the MilGrade RCD's.

**Pull request merged**